__init__.py

```diff
@@ -9,7 +9,16 @@
     from kubeflow.pipelines.components.pipelines import evaluation
 """
 
-# Import submodules to enable the convenient import patterns shown above
-# These imports ensure reliable access to submodules and better IDE support
-from . import components
-from . import pipelines
+# Import submodules to enable the convenient import patterns shown above.
+# When this module is imported as part of the installed package the relative
+# imports below work naturally. During certain tooling scenarios (e.g. pytest
+# collecting this file directly from the repo root) Python treats this module
+# as top-level and `__package__` is empty, so we fall back to absolute imports.
+if __package__ in (None, ""):
```

Comment:
this seems like a bit of an anti-pattern (conditional packaging).  I think the problem this is addressing (running pytest tests) can be fixed via a pytest.ini file and/or updates to pyproject.toml